projects/com.oracle.truffle.llvm.types/src/com/oracle/truffle/llvm/types/memory/LLVMHeap.java

```diff
@@ -66,22 +61,10 @@ public static void freeMemory(LLVMAddress addr) {
         UNSAFE.freeMemory(extractAddrNullPointerAllowed(addr));
     }
 
-    // setters
-
     public static void memCopy(LLVMAddress target, LLVMAddress source, long length) {
         long targetAddress = extractAddrNullPointerAllowed(target);
         long sourceAddress = extractAddrNullPointerAllowed(source);
```

Comment:
If I remember correctly, at least one GCC test case fails if we do not allow null pointers at this location. Both the C standard library and LLVM intrinsic do not state whether null pointers are allowed, so behavior is undefined like in the other cases that do not allow null pointers. We still have the assertions in the other cases, since it is convenient to detect NULL pointers without crashing the JVM.